tests/js/open-behavior-tests.js

```diff
@@ -356,6 +356,68 @@ module.exports = {
     }
   },
 
+  testNewFile_downloadBeforeOpen_openLocalOnTimeOut_existing: async function () {
+    // This is a regression test.  If one were to logout before the openLocalRealm timeout
+    // then the timeout would try to open a local realm with the timed out user
+
+    // 1. Add data to server Realm from User
+    // 2. Open Realm again with User
+    // 3. Logout User and run timeout
+    // 4. It should not crash
+
+    const app = new Realm.App(APP_CONFIG);
+    const partitionValue = Utils.genPartition();
+
+    const returningUserCredentials = await Utils.getRegisteredEmailPassCredentials(app);
+
+    {
+      // Add data to the realm with a different user
+      const user = await app.logIn(returningUserCredentials);
+      const config = {
+        schema: [schemas.DogForSync],
+        sync: {
+          user,
+          partitionValue,
+          _sessionStopPolicy: "immediately",
+        },
+      };
+
+      const realm = await Realm.open(config);
+
+      realm.write(() => {
+        realm.create(schemas.DogForSync.name, { _id: new ObjectId(), name: "Lola" });
+      });
+
+      await realm.syncSession.uploadAllLocalChanges();
+
+      realm.close();
+      await user.logOut();
+    }
+
+    {
+      const user = await app.logIn(returningUserCredentials);
+      const config = {
+        schema: [schemas.DogForSync],
+        sync: {
+          user,
+          partitionValue,
+          _sessionStopPolicy: "immediately",
+          existingRealmFileBehavior: {
+            type: "downloadBeforeOpen",
+            timeOut: 1000,
+            timeOutBehavior: "openLocalRealm",
+          },
+        },
+      };
+      await Realm.open(config);
```

Comment:
As you're awaiting the promise returned by realm.open before you log out, why would this trigger the original bug?
Did this throw before you did the fix?